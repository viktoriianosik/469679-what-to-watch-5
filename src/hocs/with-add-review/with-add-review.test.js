import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withAddReview from "./with-add-review";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withAddReview(MockComponent);

describe(`withAddReview is rendered correctly`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let withAddReviewComponent = null;

  beforeEach(() => {
    store = mockStore({
    });

    store.dispatch = jest.fn();

    withAddReviewComponent = renderer.create(
        <Provider store={store}>
          <MockComponentWrapped
            onSubmit={() => {}}
          >
            <React.Fragment />
          </MockComponentWrapped>
        </Provider>
    );
  });

  it(`Should withAddReviewComponent connected to store render correctly`, () => {
    expect(withAddReviewComponent.toJSON()).toMatchSnapshot();
  });
});
