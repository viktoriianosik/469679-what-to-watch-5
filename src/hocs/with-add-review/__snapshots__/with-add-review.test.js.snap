// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAddReview is rendered correctly Should withAddReviewComponent connected to store render correctly 1`] = `<div />`;
